### PR TITLE
Cloudant Geo property in design_document.py

### LINKED ISSUE #188: Geospatial index management
Support geospatial index management.
`st_indexes` property for the locally cached DesignDocument in `design_document.py`.
Parent item: #137

### PR DESCRIPTION
**Pull request by emlaver**
## What

Support geospatial indexes (Cloudant Geo) with ``st_indexes`` property in design document module.
Documentation and examples are included with the property accessor.

## How

- Added Cloudant Geo st_indexes property with documentation and example

## Testing

Added test case in DesignDocumentTests.
Created local sphinx build to review documentation.

## Issues
fixes #188

### CODE REVIEW

File: src/cloudant/design_document.py [hunk 1]
```diff
@@ -49,6 +49,45 @@ def __init__(self, database, document_id=None):
...
+                    'index': 'function(doc) { '
+                             'if (doc.geometry && doc.geometry.coordinates) { '
+                             'st_index(doc.geometry);}} '
+                }
+            }
+            ddoc.save()
+
+        Once the Cloudant Geo index is saved to the remote database, you can
+        query the index with a GET request to the ``_geo`` endpoint.
+        Example query: ``/$DATABASE/_design/doc/_geo/geoidx?<query-parameters>``
```
Comment (ricellis) [inline src/cloudant/design_document.py]: I appreciate that you added the test after my last comment which would let people fine how to do this, I do think the doc should help guide them to what is necessary.
Perhaps we could just link to http://python-cloudant.readthedocs.io/en/latest/getting_started.html#endpoint-access to help them understand that to do the query they will need to form the request themselves.

Comment (emlaver) [inline src/cloudant/design_document.py]: Added the link in 64b17f0.

File: src/cloudant/design_document.py [hunk 2]
```diff
@@ -49,6 +49,49 @@ def __init__(self, database, document_id=None):
...
+                             'if (doc.geometry && doc.geometry.coordinates) { '
+                             'st_index(doc.geometry);}} '
+                }
+            }
+            ddoc.save()
+
+        Once the Cloudant Geo index is saved to the remote database, you can
+        query the index with a GET request.  To issue a request against the
+        ``_geo`` endpoint, see the steps outlined in the `endpoint access
+        documentation http://python-cloudant.readthedocs.io/en/latest/
```
Comment (ricellis) [inline src/cloudant/design_document.py]: is there a way to point to this within the documentation rather than using the external URL?

Comment (emlaver) [inline src/cloudant/design_document.py]: Yes, I've fixed this in https://github.com/cloudant/python-cloudant/pull/192/commits/e32424108123986681d9e27862c744dd092ff76b#diff-f2994c8b7af6529726f4b4fbdd48e119R81.

File: src/cloudant/design_document.py [hunk 3]
```diff
@@ -49,6 +49,46 @@ def __init__(self, database, document_id=None):
             self.setdefault(prop, dict())
 
     @property
+    def st_indexes(self):
+        """
+        Provides an accessor property to the Cloudant geospatial
```
Comment (rhyshort) [inline src/cloudant/design_document.py]: Slightly picky, but don't we refer to `Cloudant Geospatial` with the capital G on Geospatial?

Comment (emlaver) [inline src/cloudant/design_document.py]: Yes - fixed in 0ff57ce.

**Comment by rhyshort** [comment]
+1

**Comment by ricellis** [comment]
+1

**Pull request merged**